OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/optimobile/AnalyticsContract.java

```diff
@@ -119,28 +119,33 @@ public void run() {
                 AnalyticsUploadHelper helper = new AnalyticsUploadHelper();
                 AnalyticsUploadHelper.Result result = helper.flushEvents(mContext);
 
-                if (result == AnalyticsUploadHelper.Result.SUCCESS) {
+                if (result != AnalyticsUploadHelper.Result.FAILED_RETRY_LATER) {
                     return;
                 }
                 // On failures, fall through to scheduling a background sync
             }
 
-            // Schedule a sync
-            Constraints taskConstraints = new Constraints.Builder()
-                    .setRequiredNetworkType(NetworkType.CONNECTED)
-                    .build();
+            AnalyticsContract.sheduleEventSync(mContext);
+        }
+    }
+
+    static class FlushEventsRunnable implements Runnable {
+        private final Context mContext;
+
+        FlushEventsRunnable(Context context) {
+            this.mContext = context.getApplicationContext();
+        }
 
-            OneTimeWorkRequest.Builder taskBuilder = new OneTimeWorkRequest.Builder(AnalyticsUploadWorker.class)
-                    .setConstraints(taskConstraints);
+        @Override
+        public void run() {
+            AnalyticsUploadHelper helper = new AnalyticsUploadHelper();
+            AnalyticsUploadHelper.Result result = helper.flushEvents(mContext);
 
-            if (BuildConfig.DEBUG) {
-                taskBuilder.setInitialDelay(10, TimeUnit.SECONDS);
-            } else {
-                taskBuilder.setInitialDelay(1, TimeUnit.MINUTES);
+            if (result != AnalyticsUploadHelper.Result.FAILED_RETRY_LATER) {
+                return;
             }
 
-            WorkManager.getInstance(mContext).enqueueUniqueWork(AnalyticsUploadWorker.TAG,
-                    ExistingWorkPolicy.REPLACE, taskBuilder.build());
+            AnalyticsContract.sheduleEventSync(mContext);
```

Comment:
It looks like the `AnalyticsUploadWorker` implmentation was repeated here. Can we reuse it?

Reply:
Worker is smth that runs on the bg and periodically tries to sync. In FlushEventsRunnable we try to flush (also), but if if that fails we schedule the worker. 

So, 2 lines are repeated, but overall i think they are diff things